go/worker/compute/merge/committee/node.go

```diff
@@ -273,8 +274,9 @@ func (n *Node) HandleNewBlockLocked(blk *block.Block) {
 // HandleResultsFromExecutorWorkerLocked processes results from an executor worker.
 // Guarded by n.commonNode.CrossNode.
 func (n *Node) HandleResultsFromExecutorWorkerLocked(spanCtx opentracing.SpanContext, commit *commitment.ExecutorCommitment) {
-	// TODO: Context.
-	if err := n.handleResultsLocked(context.TODO(), commit); err != nil {
+	call := func() error { return n.handleResultsLocked(n.roundCtx, commit) }
+	bkoff := backoff.WithContext(backoff.NewExponentialBackOff(), n.roundCtx)
+	if err := backoff.Retry(call, bkoff); err != nil {
```

Comment:
This probably needs to be changed to perform retry in an async fashion, otherwise this can block all processing and the error condition will never resolve (as the lock is held all the time. So probably just spawn retry in a separate go routine and make sure to take the lock in the call function each time.